konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/data/helpers/ProtobufUtils.java

```diff
@@ -93,8 +106,130 @@ else if (value.type().equals(ValueType.NDARRAY)) {
+                            setListValue(toAddGen).
+                            setListTypeValue(ValueType.STRING.ordinal()).
+                            setTypeValue(ValueType.LIST.ordinal()).
+                            build();
+                }
+                if (lv.elementType() == ValueType.IMAGE) {
+                    List<Image> images = (List<Image>)nextItem.getValue().get();
+                    List<DataProtoMessage.Image> pbImages = new ArrayList<>();
+                    for (val image : images) {
+                        Png p = image.getAs(Png.class);
+                        byte[] imageBytes = p.getBytes();
+                        DataProtoMessage.Image pbImage = DataProtoMessage.Image.newBuilder().
+                                addData(ByteString.copyFrom(imageBytes)).
+                                build();
+                        pbImages.add(pbImage);
+                    }
+
+                    DataProtoMessage.ImageList toAdd = DataProtoMessage.ImageList.newBuilder().addAllList(pbImages).build();
+                    DataProtoMessage.List toAddGen = DataProtoMessage.List.newBuilder().setImList(toAdd).build();
+                    item = DataProtoMessage.DataScheme.newBuilder().
+                            setListValue(toAddGen).
+                            setListTypeValue(ValueType.IMAGE.ordinal()).
+                            setTypeValue(ValueType.LIST.ordinal()).
+                            build();
+                }
+                if (lv.elementType() == ValueType.NDARRAY) {
+                    List<NDArray> arrays = (List<NDArray>)nextItem.getValue().get();
+                    List<DataProtoMessage.NDArray> pbArrays = new ArrayList<>();
+                    for (val arr : arrays) {
+                        SerializedNDArray sn = arr.getAs(SerializedNDArray.class);
+                        List<ByteString> byteStringList = ndArrayToByteStringList(sn);
+                        DataProtoMessage.NDArray pbNDArray = DataProtoMessage.NDArray.newBuilder().
+                                addAllShape(Arrays.asList(ArrayUtils.toObject(sn.getShape()))).
+                                addAllArray(byteStringList).
+                                setType(DataProtoMessage.NDArray.ValueType.FLOAT).
+                                build();
+                        pbArrays.add(pbNDArray);
+                    }
+                    DataProtoMessage.NDArrayList toAdd = DataProtoMessage.NDArrayList.newBuilder().addAllList(pbArrays).build();
+                    DataProtoMessage.List toAddGen = DataProtoMessage.List.newBuilder().setNdList(toAdd).build();
+                    item = DataProtoMessage.DataScheme.newBuilder().
+                            setListValue(toAddGen).
+                            setListTypeValue(ValueType.NDARRAY.ordinal()).
+                            setTypeValue(ValueType.LIST.ordinal()).
+                            build();
+                }
```

Comment:
Let's do i`f(...){ } else if(...){ } else { throw new RuntimeException("List type not yet supported/implemented: " + ...)}`
At least if we add new dtype and forget to update this method we get to know immediately rather than get our data going missing silently (which could be quite difficult to debug in say a distributed environment)


Also (separate issue) - no `List<BoundingBox>` yet.

Reply:
Fixed both.